patches/minecraft/net/minecraft/world/entity/player/Player.java.patch

```diff
@@ -258,7 +258,7 @@
 -                     for(LivingEntity livingentity : this.f_19853_.m_45976_(LivingEntity.class, p_36347_.m_20191_().m_82377_(1.0D, 0.25D, 1.0D))) {
 -                        if (livingentity != this && livingentity != p_36347_ && !this.m_7307_(livingentity) && (!(livingentity instanceof ArmorStand) || !((ArmorStand)livingentity).m_31677_()) && this.m_20280_(livingentity) < 9.0D) {
 +                     for(LivingEntity livingentity : this.f_19853_.m_45976_(LivingEntity.class, this.m_21120_(InteractionHand.MAIN_HAND).getSweepHitBox(this, p_36347_))) {
-+                        if (livingentity != this && livingentity != p_36347_ && !this.m_7307_(livingentity) && (!(livingentity instanceof ArmorStand) || !((ArmorStand)livingentity).m_31677_()) && this.canHit(livingentity, 0)) { // Original check was dist < 3, range is 3, so vanilla used padding=0
++                        if (livingentity != this && livingentity != p_36347_ && !this.m_7307_(livingentity) && (!(livingentity instanceof ArmorStand) || !((ArmorStand)livingentity).m_31677_()) && this.canReach(livingentity, 0)) { // Original check was dist < 3, range is 3, so vanilla used padding=0
```

Comment:
Is that significant?  I.e. do we care about a slight disparity on sweeping edge attacks being valid?  We _might_ be able to raise this as a vanilla bug, because entities within the sweeping hitbox aren't valid for attacking unless their center is valid.

For standard entities, the difference is (at most) the radius of their hitbox.  For large entities, we end up with the same thing as in the below discussion, where it probably makes more sense to make it work correctly as it won't impact vanilla.